app/channel.go

```diff
@@ -197,6 +197,60 @@ func WaitForChannelMembership(channelId string, userId string) {
 	}
 }
 
+func CreateGroupChannel(userIds []string) (*model.Channel, *model.AppError) {
+	if len(userIds) > model.CHANNEL_GROUP_MAX_USERS || len(userIds) < model.CHANNEL_GROUP_MIN_USERS {
+		return nil, model.NewAppError("CreateGroupChannel", "api.channel.create_group.bad_size.app_error", nil, "", http.StatusBadRequest)
+	}
+
+	var users []*model.User
+	if result := <-Srv.Store.User().GetProfileByIds(userIds, true); result.Err != nil {
+		return nil, result.Err
+	} else {
+		users = result.Data.([]*model.User)
+	}
+
+	if len(users) != len(userIds) {
+		return nil, model.NewAppError("CreateGroupChannel", "api.channel.create_group.bad_user.app_error", nil, "user_ids="+model.ArrayToJson(userIds), http.StatusBadRequest)
+	}
+
+	group := &model.Channel{
+		Name:        model.GetGroupNameFromUserIds(userIds),
+		DisplayName: model.GetGroupDisplayNameFromUsers(users, true),
+		Type:        model.CHANNEL_GROUP,
+	}
+
+	if result := <-Srv.Store.Channel().Save(group); result.Err != nil {
+		if result.Err.Id == store.CHANNEL_EXISTS_ERROR {
+			return result.Data.(*model.Channel), nil
+		} else {
+			return nil, result.Err
+		}
+	} else {
+		channel := result.Data.(*model.Channel)
```

Comment:
I could modify it to not return immediately if the channel exists and to still try to create the channel members (or check if the exist first). That way even if that sort of rare failure does happen, the next time someone tries to create the GM it should just work